src/deprecated.jl

```diff
@@ -15,3 +15,48 @@ Base.@deprecate_binding parseatom Meta.parseatom false
 Base.@deprecate_binding parseall Meta.parseall false
 import UUIDs
 Base.@deprecate_binding uuid5 UUIDs.uuid5 true
+
+# https://github.com/JuliaLang/julia/pull/46104
+# reverted in https://github.com/JuliaLang/julia/pull/52010
+if VERSION < v"1.10.0-DEV.1404" ||
+    (VERSION >= v"1.10-rc2" && VERSION < v"1.11.0-") ||
+    VERSION >= v"1.11.0-DEV.924"
```

Comment:
```suggestion
if VERSION < v"1.10.0-DEV.1404"
```
To stop adding methods to `Base.sort` in future Julia versions while keeping `sort` working on general iterables on Julia < 1.10 with Compat (where people might already be using it).

Ref. https://github.com/JuliaLang/Compat.jl/pull/811#issuecomment-1831397828